File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        left = 0
        right = len(numbers) - 1

        while left <= right:
            x = numbers[left] + numbers[right]
            if target == x:
                return [left+1, right+1]
            
            if x < target:
                left += 1
            else:
                right -= 1
        
        return []
        

File: Data Structures & Algorithms/search-2d-matrix/submission-1.swift
class Solution {
    func searchMatrix(_ matrix: [[Int]], _ target: Int) -> Bool {
        let n = matrix[0].count
        var left = 0
        var right = (matrix.count * n) - 1 // (m * n) - 1
    
        while left <= right{
            let mid = left + (right - left) / 2
            let a = mid / n
            let b = mid % n
            if matrix[a][b] == target{
                return true
            }
            if matrix[a][b] < target{
                left = mid + 1
            }else{
                right = mid - 1
            }
        }
        return false
    }
}
